### DISCUSSION TITLE
Github actions services available on others VMs ?

### DISCUSSION THREAD

**Discussion by sabderemane** [community/community#25049] [Actions]
<p>Hi,</p>
<p>I’ve tried to setup a simple action with postgres service, by curiosity, with a matrix strategy to lauch the workflow also on windows VM but I’ve seen this error : </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/5/5ce61ee8c27df93d0314cbd21b32333a97ca1ebe.png" data-download-href="/uploads/short-url/dfOXrcVjmnQQYaYchIkwsWZQiaq.png?dl=1" title="Capture d’écran 2019-11-16 à 14.13.07.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/17890338/181088566-0261d277-4d20-4c59-96a9-317daac0adf9.png" alt="Capture d’écran 2019-11-16 à 14.13.07.png" data-base62-sha1="dfOXrcVjmnQQYaYchIkwsWZQiaq" width="690" height="199" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/5/5ce61ee8c27df93d0314cbd21b32333a97ca1ebe_2_690x199.png, https://aws1.discourse-cdn.com/github/optimized/2X/5/5ce61ee8c27df93d0314cbd21b32333a97ca1ebe_2_1035x298.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/5/5ce61ee8c27df93d0314cbd21b32333a97ca1ebe_2_1380x398.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/5/5ce61ee8c27df93d0314cbd21b32333a97ca1ebe_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Capture d’écran 2019-11-16 à 14.13.07.png</span><span class="informations">1460×422 60.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I know services uses Docker container and I’m guessing it’s a bit tricky in others VMs. Github actions services will support other virtual machine like Windows later or it’s not planned at all ?</p>
<p>Thanks in advance,</p>

**Comment by BrightRan** [comment, marked as answer]
<p>Hi <a class="mention" href="/sabderemane">@sabderemane</a> ,</p>
<p>Currently, Docker container actions can only execute in the GitHub-hosted Linux environment, and is not supported on other environments (such as Windows and MacOS).</p>
<p>More details please reference here: <a href="https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fhelp.github.com%2Fen%2Factions%2Fautomating-your-workflow-with-github-actions%2Fabout-actions%23types-of-actions&amp;data=02%7C01%7Cv-brira%40microsoft.com%7C0a96d1fcbed44fed7c5f08d76c09fde4%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637096662480955752&amp;sdata=nQ2NIdEu7wq0P3FIfC2vLa5is%2BYBuh8Z272AGPrtGG0%3D&amp;reserved=0" rel="nofollow noopener">https://help.github.com/en/actions/automating-your-workflow-with-github-actions/about-actions#types-of-actions</a></p>
<p>We notice that some other users also had reported the same question, we had reported this as a feature request to the appropriate engineering team.</p>

**Comment by ruudboon** [comment]
<p>Would be nice if the service support can be mentioned here as well. <br>
 <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idservices" rel="nofollow noopener">https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idservices</a></p>
